analytic_engine/src/memtable/skiplist/iter.rs

```diff
@@ -162,6 +164,12 @@ impl<A: Arena<Stats = BasicStats> + Clone + Sync + Send> ColumnarIterImpl<A> {
         }
 
         if num_rows > 0 {
+            if let Some(deadline) = self.deadline {
+                if deadline.saturating_elapsed().is_zero() {
+                    return IterTimeout {}.fail();
+                }
+            }
```

Comment:
Maybe it is more readable to define a method called `check_deadline` for this code block.

Reply:
Fixed